abn/application/applicationmap.go

```diff
@@ -239,10 +239,18 @@ func (m *ThreadSafeApplicationMap) BatchedWrite(a *Application) error {
 	lastWrite, ok := m.lastWriteTimes[a.Name]
 	if !ok || lastWrite == nil {
 		// no record of the application ever being written; write it now
-		m.Write(a)
+		err := m.Write(a)
+		if err != nil {
```

Comment:
Never ignore real world errors in actual code.

Reply:
So this is going to require some thinking. Errors are sometimes ignored for a reason. In this case, if `Write` fails, it fails. Agree that logging the error is probably a good idea. But I don't think we always need to return it.

Reviewed all uses of this method (`BatchedWrite`) and it is ok to return it.

Reply:
This is exactly right. Whenever you ignore errors, do so explicitly, and not silently.